java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -220,49 +220,14 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_makeList(JNIEnv *env, j
 JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_fromScalar(JNIEnv *env, jclass,
                                                                     jlong j_scalar,
                                                                     jint row_count) {
-  using ScalarType = cudf::scalar_type_t<cudf::size_type>;
   JNI_NULL_CHECK(env, j_scalar, "scalar is null", 0);
   try {
     cudf::jni::auto_set_device(env);
     auto scalar_val = reinterpret_cast<cudf::scalar const *>(j_scalar);
-    auto dtype = scalar_val->type();
-    cudf::mask_state mask_state =
-        scalar_val->is_valid() ? cudf::mask_state::UNALLOCATED : cudf::mask_state::ALL_NULL;
     std::unique_ptr<cudf::column> col;
-    if (dtype.id() == cudf::type_id::LIST) {
-      // Neither 'cudf::make_empty_column' nor 'cudf::make_column_from_scalar' supports
-      // LIST type for now (https://github.com/rapidsai/cudf/issues/8088), so the list
-      // precedes the others and takes care of the empty column itself.
-      auto s_list = reinterpret_cast<cudf::list_scalar const *>(scalar_val);
-      cudf::column_view s_val = s_list->view();
-
-      // Offsets: [0, list_size, list_size*2, ..., list_szie*row_count]
-      auto zero = cudf::make_numeric_scalar(cudf::data_type(cudf::type_id::INT32));
-      auto step = cudf::make_numeric_scalar(cudf::data_type(cudf::type_id::INT32));
-      zero->set_valid(true);
-      step->set_valid(true);
-      static_cast<ScalarType *>(zero.get())->set_value(0);
-      static_cast<ScalarType *>(step.get())->set_value(s_val.size());
-      std::unique_ptr<cudf::column> offsets = cudf::sequence(row_count + 1, *zero, *step);
-      // Data:
-      // Builds the data column by leveraging `cudf::concatenate` to repeat the 's_val'
-      // 'row_count' times, because 'cudf::make_column_from_scalar' does not support list
-      // type.
-      // (Assumes the `row_count` is not big, otherwise there would be a performance issue.)
-      // Checks the `row_count` because `cudf::concatenate` does not support no rows.
-      auto data_col = row_count > 0
-          ? cudf::concatenate(std::vector<cudf::column_view>(row_count, s_val))
-          : cudf::empty_like(s_val);
```

Comment:
This original code looks like it would handle the case where row_count == 0 but the type is a LIST of STRUCT of something.  The new cudf code [seems to barf on empty STRUCTS](https://github.com/rapidsai/cudf/pull/8185/files#diff-1140f9642e1a1dcc42c6830acb3f3681eb71f63f36cd7444cfb692e6c487777bR101):
```
if (size == 0) CUDF_FAIL("0-length struct column is unsupported.");
```

Reply:
I believe this is only for the struct scalar case, which we have not declared to support.

1. We have the test for creating an empty column from the list of struct scalar, and it passes. The code is here [ColumnVectorTest.java#L1531](https://github.com/rapidsai/cudf/blob/branch-21.06/java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java#L1531)
```Scala
  @Test
  void testFromScalarListOfStruct() {
    ...
      // empty row
      try (ColumnVector ret = ColumnVector.fromScalar(s, 0)) {
        assertEquals(ret.getRowCount(), 0);
        assertEquals(ret.getNullCount(), 0);
      }
    }
  }
```
2. The code handling the case you mentioned should be here [lists_column_factories.cu#L40](https://github.com/rapidsai/cudf/blob/branch-21.06/cpp/src/lists/lists_column_factories.cu#L40), the same as we did before.

Reply:
It seems that `column_from_scalar_dispatch<struct_view>` will be called only if the root type is Struct. (will not be called for Structs in Lists).